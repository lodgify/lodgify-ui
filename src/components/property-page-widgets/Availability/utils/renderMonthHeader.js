import React from 'react';
import moment from 'moment';

import { buildKeyFromStrings } from 'utils/build-key-from-strings';

const buildWeekDayMappingMarkup = () => (
  <ul className="DayPicker_weekHeader_ul">
    {moment.weekdays(true).map((weekday, index) => (
      <li
        className="DayPicker_weekHeader_li"
        key={buildKeyFromStrings(weekday, index)}
      >
        <small>
          {moment()
            .day(weekday)
            .format('dd')}
        </small>
      </li>
    ))}
  </ul>
);

/**
 * @param   {string} localeCode
 * @return  {Function}
 */
// eslint-disable-next-line react/prop-types
export const renderMonthHeader = localeCode => ({ month }) => {
  month.locale(localeCode);

  return (
    <div className="CalendarMonth_caption">
      <strong>{month.format('MMMM YYYY')}</strong>
      <div className="DayPicker_weekHeaders">
        <div className="DayPicker_weekHeader">
          {buildWeekDayMappingMarkup()}
        </div>
      </div>
    </div>
  );
};
