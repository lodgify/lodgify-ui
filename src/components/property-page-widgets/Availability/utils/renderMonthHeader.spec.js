import moment from 'moment';
import { shallow } from 'enzyme';

import { renderMonthHeader } from './renderMonthHeader';

const today = moment();

describe('renderMonthHeader', () => {
  const getMonthHeader = () =>
    shallow(
      renderMonthHeader({
        month: today,
      })
    );

  it('should correctly render 7 list items for each day', () => {
    const wrapper = getMonthHeader().find('li');

    expect(wrapper).toHaveLength(7);
    expect(
      wrapper
        .at(0)
        .find('small')
        .text()
    ).toContain('Su');
    expect(
      wrapper
        .at(6)
        .find('small')
        .text()
    ).toContain('Sa');
  });

  it('should correctly render the month title', () => {
    const wrapper = getMonthHeader().find('strong');

    expect(wrapper.text()).toBe(today.format('MMMM YYYY'));
  });
});
