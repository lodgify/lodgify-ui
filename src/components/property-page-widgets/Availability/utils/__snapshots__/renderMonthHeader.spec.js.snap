// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderMonthHeader the returned function should return the correct structure 1`] = `
<div
  className="CalendarMonth_caption"
>
  <strong>
    11ì›” 2018
  </strong>
  <div
    className="DayPicker_weekHeaders"
  >
    <div
      className="DayPicker_weekHeader"
    >
      <ul
        className="DayPicker_weekHeader_ul"
      >
        <li
          className="DayPicker_weekHeader_li"
          key="0"
        >
          <small>
            Su
          </small>
        </li>
        <li
          className="DayPicker_weekHeader_li"
          key="1"
        >
          <small>
            Mo
          </small>
        </li>
        <li
          className="DayPicker_weekHeader_li"
          key="2"
        >
          <small>
            Tu
          </small>
        </li>
        <li
          className="DayPicker_weekHeader_li"
          key="3"
        >
          <small>
            We
          </small>
        </li>
        <li
          className="DayPicker_weekHeader_li"
          key="4"
        >
          <small>
            Th
          </small>
        </li>
        <li
          className="DayPicker_weekHeader_li"
          key="5"
        >
          <small>
            Fr
          </small>
        </li>
        <li
          className="DayPicker_weekHeader_li"
          key="6"
        >
          <small>
            Sa
          </small>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
